wrappers/csharp/Managed.cs

```diff
@@ -862,6 +863,43 @@ public static string DecryptWithPasswordAsString(string b64data, string password
             return DecryptWithPasswordAsUtf8String(b64data, password, iterations);
         }
 
+        /// <summary>
+        /// Join multiple shares to regenerate a shared secret.
+        /// </summary>
+        /// <param name="shares">This is a 2-dimensional array representing the shares.</param>
+        /// <returns>The output buffer with the shared secret.</returns>
+        public static byte[] JoinShares(byte[][] shares)
+        {
+            if (shares == null || shares.Length == 0 || !SharesAreValid(shares))
+            {
+                throw new DevolutionsCryptoException(ManagedError.InvalidParameter);
+            }
+
+            try
+            {
+                int nbShares = shares.Length;
+                int sharesLength = shares[0].Length;
+                int secretLength = (int)Native.JoinSharesSizeNative((UIntPtr)sharesLength);
+                byte[] secret = new byte[secretLength];
+                byte[][] sharesTemp = new byte[nbShares][];
+                Array.Copy(shares, sharesTemp, shares.Length);
+                IntPtr[] pointers = GetArrayReferences(ref shares, nbShares, secretLength);
+                Array.Copy(sharesTemp, shares, sharesTemp.Length);
+                long result = Native.JoinSharesNative((UIntPtr)nbShares, (UIntPtr)sharesLength, pointers, secret, (UIntPtr)secretLength);
+
+                return secret;
```

Comment:
1. You should free the pointers after you're done with it.
2. The native errors aren't properly handled

Reply:
Pointers are released inside the GetArrayReferences and InitializeArray methods. In the example, It is the handles [i] .Free (), which must be done

Reply:
Free() call should still be done *after* they are used, which means after the Native.JoinSharesNative() call